api/v1beta1/defaults.go

```diff
@@ -67,6 +67,11 @@ func SetDefaults_AWSClusterSpec(s *AWSClusterSpec) { //nolint:golint,stylecheck
 	} else if s.ControlPlaneLoadBalancer.Scheme != nil && s.ControlPlaneLoadBalancer.Scheme.String() == ClassicELBSchemeIncorrectInternetFacing.String() {
 		s.ControlPlaneLoadBalancer.Scheme = &ClassicELBSchemeInternetFacing
 	}
+
+	// Set Default value of HealthCheckProtocol if not set
+	if s.ControlPlaneLoadBalancer != nil && s.ControlPlaneLoadBalancer.HealthCheckProtocol == nil {
```

Comment:
This is defaulting webhook, so stores the change back to api serves, which as per our discussions makes this a breaking change.

Reply:
so shall we add a check for immutability in cluster webhook?

Reply:
so shall we remove this check from here and keep it in loadbalancer code only?

Reply:
How can we do [this](https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/3124#issuecomment-1031140120) here

Reply:
Would defaulting [here](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/638278a552ff92c992e57a926f4598bc6bcccaec/pkg/cloud/services/elb/loadbalancer.go#L789) not suffice?